src/vm/runners/cairo_runner.rs

```diff
@@ -443,6 +445,29 @@ impl CairoRunner {
         Ok(())
     }
 
+    pub fn get_execution_resources(
+        &self,
+        vm: &VirtualMachine,
+    ) -> Result<ExecutionResources, TraceError> {
+        let n_steps = match self.original_steps {
+            Some(x) => x,
+            None => vm.trace.as_ref().ok_or(TraceError::TraceNotEnabled)?.len(),
+        };
```

Comment:
Is it OK for this to fail if both `vm.trace` and `self.original_steps` are `None`?